packages/xpetra/src/MultiVector/Xpetra_TpetraMultiVector_def.hpp

```diff
@@ -358,107 +390,76 @@ namespace Xpetra {
-  template <class Scalar, class LocalOrdinal, class GlobalOrdinal, class Node>
-  const Tpetra::MultiVector< Scalar,LocalOrdinal, GlobalOrdinal, Node> & toTpetra(const MultiVector< Scalar,LocalOrdinal, GlobalOrdinal, Node> &x) {
-    typedef TpetraMultiVector< Scalar, LocalOrdinal, GlobalOrdinal, Node > TpetraMultiVectorClass;
-      XPETRA_DYNAMIC_CAST(const TpetraMultiVectorClass, x, tX, "toTpetra");
-      return *tX.getTpetra_MultiVector();
+  TpetraMultiVector<Scalar,LocalOrdinal,GlobalOrdinal,Node>::    
+  getLocalView () const {
+    return this->MultiVector< Scalar, LocalOrdinal, GlobalOrdinal, Node >::template getLocalView<TargetDeviceType>();
   }
-
-  template <class Scalar, class LocalOrdinal, class GlobalOrdinal, class Node>
-  Tpetra::MultiVector< Scalar,LocalOrdinal, GlobalOrdinal, Node> & toTpetra(MultiVector< Scalar,LocalOrdinal, GlobalOrdinal, Node> &x) {
-    typedef TpetraMultiVector< Scalar, LocalOrdinal, GlobalOrdinal, Node > TpetraMultiVectorClass;
-      XPETRA_DYNAMIC_CAST(      TpetraMultiVectorClass, x, tX, "toTpetra");
-      return *tX.getTpetra_MultiVector();
+#endif
+  
+  template<class Scalar, class LocalOrdinal, class GlobalOrdinal, class Node>
+  typename TpetraMultiVector<Scalar,LocalOrdinal,GlobalOrdinal,Node>::dual_view_type::t_host_um 
+  TpetraMultiVector<Scalar,LocalOrdinal,GlobalOrdinal,Node>::    
+  getHostLocalView () const {
+    return subview(vec_->template getLocalView<typename TpetraMultiVector<Scalar,LocalOrdinal,GlobalOrdinal,Node>::dual_view_type::host_mirror_space> (),
+                   Kokkos::ALL(), Kokkos::ALL());
   }
-  //
-
-
-  // Things we actually need
-  template <class Scalar, class LocalOrdinal, class GlobalOrdinal, class Node>
-  RCP<MultiVector<Scalar, LocalOrdinal, GlobalOrdinal, Node > > toXpetra(RCP<Tpetra::MultiVector< Scalar, LocalOrdinal, GlobalOrdinal, Node > > vec) {
-    if (!vec.is_null())
-      return rcp(new TpetraMultiVector<Scalar, LocalOrdinal, GlobalOrdinal, Node >(vec));
 
-    return Teuchos::null;
+  template<class Scalar, class LocalOrdinal, class GlobalOrdinal, class Node>
+  typename TpetraMultiVector<Scalar,LocalOrdinal,GlobalOrdinal,Node>::dual_view_type::t_dev_um 
+  TpetraMultiVector<Scalar,LocalOrdinal,GlobalOrdinal,Node>::    
+  getDeviceLocalView() const {
+    return subview(vec_->template getLocalView<typename TpetraMultiVector<Scalar,LocalOrdinal,GlobalOrdinal,Node>::dual_view_type::t_dev_um::execution_space> (),
+                   Kokkos::ALL(), Kokkos::ALL());
   }
+  
+#endif
 
-  template <class Scalar, class LocalOrdinal, class GlobalOrdinal, class Node>
-  RCP<const MultiVector<Scalar, LocalOrdinal, GlobalOrdinal, Node > > toXpetra(RCP<const Tpetra::MultiVector< Scalar, LocalOrdinal, GlobalOrdinal, Node > > vec) {
-    if (!vec.is_null())
-      return rcp(new TpetraMultiVector<Scalar, LocalOrdinal, GlobalOrdinal, Node >(vec));
-
-    return Teuchos::null;
+  /// \brief Implementation of the assignment operator (operator=);
```

Comment:
`Tpetra::MultiVector::operator=` does not do a deep copy, but `Tpetra::MultiVector::assign` does.  That is, the code looks right, but the documentation isn't quite right.